geonode/geoserver/geofence.py

```diff
@@ -0,0 +1,254 @@
+#########################################################################
+#
+# Copyright (C) 2023 OSGeo
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+#########################################################################
+
+import json
+import logging
+import requests
+from requests.auth import HTTPBasicAuth
+import urllib
+
+logger = logging.getLogger(__name__)
+
+
+class GeofenceException(Exception):
+    pass
+
+
+class Rule:
+
+    ALLOW = "ALLOW"
+    DENY = "DENY"
+    LIMIT = "LIMIT"
+
+    CM_MIXED = "MIXED"
+
+    def __init__(self, priority, workspace, layer, access: (str, bool),
+                 user=None, group=None,
+                 service=None, request=None, subfield=None,
+                 geo_limit=None, catalog_mode=None) -> None:
+        self.fields = {}
+
+        # access may be either a boolean or ALLOW/DENY/LIMIT
+        if access is True:
+            access = Rule.ALLOW
+        elif access is False:
+            access = Rule.DENY
+
+        for field, value in (
+            ('priority', priority),
+
+            ('userName', user),
+            ('roleName', group),
+
+            ('service', service),
+            ('request', request),
+            ('subfield', subfield),
+
+            ('workspace', workspace),
+            ('layer', layer),
+
+            ('access', access),
+        ):
+            if value is not None and value != '*':
+                self.fields[field] = value
+
+        limits = {}
+        for field, value in (
+            ('allowedArea', geo_limit),
+            ('catalogMode', catalog_mode),
+        ):
+            if value is not None:
+                limits[field] = value
+
+        if limits:
+            self.fields['limits'] = limits
+
+    def get_object(self):
+        logger.debug(f"Creating Rule object: {self.fields}")
+        return {'Rule': self.fields}
+
+
+class Batch:
+    def __init__(self, log_name=None) -> None:
+        self.operations = []
+        self.log_name = f'"{log_name}"' if log_name else ''
```

Comment:
Minor: this can also be `self.log_name = f"{log_name or ''}"`

Reply:
nope, we want quotes around the name